src/System.Private.CoreLib/shared/System/Runtime/Serialization/SerializationInfo.cs

```diff
@@ -541,103 +541,103 @@ public bool GetBoolean(string name)
         {
             Type foundType;
             object? value = GetElement(name, out foundType);
-            return ReferenceEquals(foundType, typeof(bool)) ? (bool)value : _converter.ToBoolean(value!); // if value is null To* method will either deal with it or throw
+            return ReferenceEquals(foundType, typeof(bool)) ? (bool)value! : _converter.ToBoolean(value!); // if value is null To* method will either deal with it or throw
```

Comment:
Could we add the `!` to `GetElement(…)` and make `value` be `object`?

Reply:
Good idea.

Reply:
Fixed.